tests/pass-dep/libc/pthread-threadname.rs

```diff
@@ -74,6 +74,21 @@ fn main() {
             // large enough for the thread name.
             #[cfg(target_os = "linux")]
             assert_eq!(get_thread_name(&mut buf[..15]), libc::ERANGE);
+
+            // For libc implementation for macOS it's not an error
+            // for a buffer being too short for the thread name.
+            #[cfg(any(target_os = "freebsd", target_os = "macos"))]
```

Comment:
Basically what I think we should have is a test for `#[cfg(not(any(target_os = "freebsd", target_os = "macos")))]` where we call `get_thread_name(&mut buf[..4])` and assert that we get back ERANGE.

Reply:
I wouldn't add a test right now, because I haven't verified other platforms still. So, it would be misleading even if it looks correct according to the current interpreter logic. I also have some questions regarding FreeBSD, because it supports GNU extension method to get and set thread names, but they are not used by `std` at the moment while being present in `libc`, and they might have slightly different return codes.

Reply:
Okay, had a quick look at Illumos which should be compatible with Solaris, and there's `ERANGE` too. Will do a test and write a comment.

Reply:
Added just for Solaris like OSes because Linux has it's own already.